addons/mail/models/mail_link_preview_message.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class LinkPreviewMessage(models.Model):
+    _name = "mail.link.preview.message"
+    _inherit = "bus.listener.mixin"
+    _description = "Link between link previews and messages"
+
+    link_preview_id = fields.Many2one("mail.link.preview", index=True)
```

Comment:
Individual index probably not useful with unique already covering it. At least for the first key of the unique. The second one might be still useful.